### PR TITLE
Add Dusty the bluespace cat

### PR DESCRIPTION
**Pull request by hyperioo**
## About The Pull Request

Dusty is a bluespace cat, it purrs its way into our reality when runtimes happen.

Dusty is invicible, godlike (I know some people will try to kill it just because I said that ;-;). 

It is teleported to the location of the newest runtime each time one happens, with a cooldown of 30 seconds to avoid having the cat teleported everywhere if there is runtimes spam.

Sprite by InfraredBaron
dusty

## Why It's Good For The Game

It's a **_kot_**

## Changelog
:cl: Hyperio
add: Add Dusty the bluespace cat
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/friendly/cat.dm [hunk 1]
```diff
@@ -263,3 +263,91 @@
...
+	if(istype(target_mob,/obj/machinery/bot))
+		var/obj/machinery/bot/B = target_mob
+		B.attack_generic(src,rand(melee_damage_lower,melee_damage_upper),attacktext)
+		return B
+
+/mob/living/simple_animal/cat/runtime/set_flee_target(atom/A)
+	return
+
+/mob/living/simple_animal/cat/runtime/set_flee_target(atom/A)
+	return
```
Comment (TheShown911) [inline code/modules/mob/living/simple_animal/friendly/cat.dm:346]: May have put this in twice.

**Comment by AgentMA1** [comment]
Local cat seen EVERYWHERE in the ship

**Comment by MLGTASTICa** [comment]
People are gonna be sooo confused when it randomly just appears ontop of them whenever they runtime , or mid surgery when it checks the floor for broken bones instead of the organs. Very good feature.

File: code/modules/mob/living/simple_animal/friendly/cat.dm (hunk 1 above)
Comment (hyperioo) [reply code/modules/mob/living/simple_animal/friendly/cat.dm:346]: Thanks 👍

**Comment by Firefox13** [comment]
Rather than a joke which will be literally everywhere in every round happening literally everytime, maybe you can make Dusty appears if admins spawn him in the first place?

**Comment by Wouju** [comment]
> Rather than a joke which will be literally everywhere in every round happening literally everytime, maybe you can make Dusty appears if admins spawn him in the first place?

or perhaps make the cooldown extremely long?

**Comment by Kurgis** [comment]
> Rather than a joke which will be literally everywhere in every round happening literally everytime, maybe you can make Dusty appears if admins spawn him in the first place?

Why bother going through this length of effort then?

**Comment by Cocacola4124** [comment]
singulo and biomatter food

**Review by Cocacola4124 — CHANGES_REQUESTED** [review]
Use SPAN_NOTICE, SPAN_DANGER, or SPAN_WARNING instead of \red or \blue
SPANS

**Comment by Cocacola4124** [comment]
You should also return on ex_act and singularity_act, to make them immune to gibbing from those sources.
Bluespace Gods

**Comment by hyperioo** [comment]
@bigpep Thanks for the feedback 👍

**Review by TheShown911 — COMMENTED** [review]
One dusty is way cooler than a lot of dusties, i liked the original idea more.

**Review by Firefox13 — CHANGES_REQUESTED** [review]
I agree with Shown, one cat is better than a bunch of them since it removes the individuality(and having 10 cats spawn because you did 10 things is cringe). And please, since we usually have a bunch of runtimes, make chance not 100% - like 25 or 20 can be enough, it shouldn't be guaranteed.
~~or just 1% as ultra easter egg~~

**Comment by hyperioo** [comment]
@Firefox13 I made this change because I thought you agreed with that :agony:

Screenshot_2021-05-14_16-02-33

So I revert to only one Dusty with 1 percent chance to spawn and disappear after 1 minute ?

**Comment by Firefox13** [comment]
> 
> 
> @Firefox13 I made this change because I thought you agreed with that :agony: Guess I misunderstood
> 
> Screenshot_2021-05-14_16-02-33
> 
> So I revert to only one Dusty with small percents chance to spawn and disappear after 1 minute ?

Guess [REDACTEd] misunderstood thinking that there wasn't a limit for them. Well, we can keep the amount of cats but they shouldn't be guaranteed to appear, yeah.

**Comment by hyperioo** [comment]
I've limited the spawning chance to 10% to have it between common and super rare easter egg

**Comment by Evankhell561** [comment]
> I've limited the spawning chance to 10% to have it between common (25%) and super rare easter egg (1%)

I don't like it at all, one thing is an easter egg for the players and another thing is to send a message to the admins and coders, it's not something that anyone will appreciate.

**Comment by MLGTASTICa** [comment]
> > I've limited the spawning chance to 10% to have it between common (25%) and super rare easter egg (1%)
> 
> I don't like it at all, one thing is an easter egg for the players and another thing is to send a message to the admins and coders, it's not something that anyone will appreciate.

I appreciate a cute bluespace cat and i want it merged now whenever this shitcode runtimes

**Comment by Evankhell561** [comment]
> I appreciate a cute bluespace cat and i want it merged now whenever this shitcode runtimes

fine, keep the cat but don't make him send messages to the discord that no player will be able to see.

**Comment by hyperioo** [comment]
> 
> 
> > I've limited the spawning chance to 10% to have it between common (25%) and super rare easter egg (1%)
> 
> I don't like it at all, one thing is an easter egg for the players and another thing is to send a message to the admins and coders, it's not something that anyone will appreciate.

I'm not sure I understand. Discord messages in coder-shaming are untouched and it's the tool used to spot runtimes. This cat is more like some sort of easter egg to warn players some naughty code is going on and the thing they're using may not work as intended.

**Comment by Evankhell561** [comment]
> > > I've limited the spawning chance to 10% to have it between common (25%) and super rare easter egg (1%)
> > 
> > 
> > I don't like it at all, one thing is an easter egg for the players and another thing is to send a message to the admins and coders, it's not something that anyone will appreciate.
> 
> I'm not sure I understand. Discord messages in coder-shaming are untouched and it's the tool used to spot runtimes. This cat is more like some sort of easter egg to warn players that some naughty code is running/going crazy and that what they are doing may not work as intended.

forget it then.

**Review by Firefox13 — APPROVED** [review]

**Pull request merged**